http/interceptor.go

```diff
@@ -104,11 +104,25 @@ func (i *rwInterceptor) Write(b []byte) (int, error) {
 	return i.w.Write(b)
 }
 
+// WriteString buffers the response body until the request body limit is reach or an
+// interruption is triggered, this buffer is later used to analyse the body in
+// the response processor.
+// If the body isn't accessible or the mime type isn't processable, the response
+// body is being written to the delegate response writer directly.
+func (i *rwInterceptor) WriteString(s string) (n int, err error) {
+	return i.Write([]byte(s))
+}
+
 func (i *rwInterceptor) Header() http.Header {
 	return i.w.Header()
 }
 
-var _ http.ResponseWriter = (*rwInterceptor)(nil)
+type stringResponseWriter interface {
+	http.ResponseWriter
+	io.StringWriter
```

Comment:
Do you mind pointing at an example of another similar code that implements `io.StringWriter`? This doesn't seem so bad, but I'm wondering when a caller wouldn't just call `Write([]byte(s))` themselves without inspecting the interfaces.

Reply:
This used to a thing but now I only find old references.
Like these : 
- https://go.googlesource.com/benchmarks/+/master/garbage/nethttp.go#1690
- https://github.com/golang/go/issues/39558#issuecomment-648952651

I suspect that `WriteString` on `ResponseWriter` is no longer common and that `Write([]bytes(s))` is the idiomatic way.

----------

We still have a bunch of tests to make sure that all middleware we add implements the needed interfaces. But it has been a while (±6 years) since I revisited those tests.

e.g. `CloseNotifier` is also unimplemented here but that is documented as deprecated in `http`.

So, yeah, this is ok to close for me :)
It can always be revisited if there is an actual need for this and not because of some legacy requirement.

----

Thank you for the super speedy review 🙇

Reply:
Thanks a lot for the investigation @romainmenke! Yeah let's go ahead and close for now then, if we find a strong use case for it we can reintroduce it.